### PR TITLE
port: Port to 1.20.2

### PR DESCRIPTION
**Pull request by kristofbolyai**
https://github.com/Wynntils/Artemis/pull/2095/commits/4ea0e8445a702f7e0ab7d4b809a6352f3b6579de was already reviewed. I suggest reviewing all commits 1 by 1, after the 1.20.1 commit, skipping the merge commits. Sadly 1.20.2 has had much more issues with rendering, so it was much harder to keep a clean commit history.

### CODE REVIEW

**Comment by ShadowCat117** [comment]
Cursor/input is rendered behind some input widgets. Chat tab screen, party management, color on poi creation screen for example and on bank page names you can see the cursor behind the widget.

Create/leave party button is slightly dimmed but not deactivated.

Can't find any issues functionality wise other than that

**Comment by kristofbolyai** [comment]
> Cursor/input is rendered behind some input widgets. Chat tab screen, party management, color on poi creation screen for example and on bank page names you can see the cursor behind the widget.
> 
> Create/leave party button is slightly dimmed but not deactivated.
> 
> Can't find any issues functionality wise other than that

Should be fixed now. Thank you.

File: common/src/main/java/com/wynntils/utils/mc/type/Location.java [hunk 1]
```diff
@@ -71,10 +70,6 @@ public Vec3 toVec3() {
         return new Vec3(x, y, z);
     }
 
-    public Position toPosition() {
```
Comment (magicus) [inline common/src/main/java/com/wynntils/utils/mc/type/Location.java:74]: I'm a bit surprised this was not used..? Don't you think we should keep it but return Vec3 or whatever it is they use nowadays for a basic implementation of Position?

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/utils/mc/type/Location.java:74]: We could but if the underlying class is Vec3 you can implicitly cast it

File: common/src/main/java/com/wynntils/utils/render/RenderUtils.java [hunk 2]
```diff
@@ -833,7 +833,6 @@ public static void renderVignetteOverlay(PoseStack poseStack, CustomColor color,
                 Texture.VIGNETTE.height());
 
         RenderSystem.setShaderColor(1, 1, 1, 1);
-        RenderSystem.enableBlend();
```
Comment (magicus) [inline common/src/main/java/com/wynntils/utils/render/RenderUtils.java:835]: O_o

Why did we ever do something like this? (And why did it not work?)

**Comment by magicus** [comment]
Remember to credit @wukibaka in the final merge.

File: common/src/main/java/com/wynntils/utils/mc/type/Location.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/utils/mc/type/Location.java:74]: I still think I'd like to see this kept; there is a difference in semantics that I'd like to keep, where Position is the more abstract Minecraft concept of a general (entity) position, basically a location with fractional coordinates.

File: gradle.properties [hunk 3]
```diff
@@ -4,36 +4,38 @@
...
-minecraft_version=1.19.4
+minecraft_version=1.20.2
 
 # Architectury
 # Check for latest at https://maven.architectury.dev/architectury-plugin/architectury-plugin.gradle.plugin
 architectury_plugin_version=3.4-SNAPSHOT
 # Check for latest at https://maven.architectury.dev/dev/architectury/loom/dev.architectury.loom.gradle.plugin
 architectury_loom_version=1.2-SNAPSHOT
-enabled_platforms=fabric,forge,quilt
+enabled_platforms=fabric
```
Comment (magicus) [inline gradle.properties]: Oops..?

Comment (magicus) [reply gradle.properties]: Did you try this on Quilt?

File: gradle.properties [hunk 4]
```diff
@@ -4,36 +4,38 @@
...
 # Note: Must use same version of eventbus as is used in the above forge version!
-forge_eventbus_version=6.0.3
+forge_eventbus_version=6.0.5
 
 # Quilt
-# Check for latest at https://maven.quiltmc.org/repository/release/org/quiltmc/qsl
-quilt_loader_version=0.19.3-beta.1
-qsl_version=5.0.0-beta.10+1.19.4
+# Check for latest at https://maven.quiltmc.org/repository/release/org/quiltmc/
+# FIXME: Update
```
Comment (magicus) [inline gradle.properties:36]: ... maybe not?

File: settings.gradle [hunk 5]
```diff
@@ -9,7 +9,8 @@ pluginManagement {
 
 include("common")
 include("fabric")
-include("quilt")
+// FIXME: Include loaders
```
Comment (magicus) [inline settings.gradle:12]: Here as well.

**Comment by magicus** [comment]
It's always quite a bit to review minecraft ports like this. Overall, I think it looked quite straightforward.

You need to fix Quilt. (And add back Forge, and verify that it really works)

Also, I'd like to see Location.toPosition() restored.

Apart from that, it all looks good!

**Review by magicus — APPROVED** [review]

**Comment by kristofbolyai** [comment]
> Remember to credit @wukibaka in the final merge.

I think git does automatically.

File: gradle.properties (hunk 3 above)
Comment (kristofbolyai) [reply gradle.properties]: Quilt was not out yet (forge neither, but I added it back). This will be fixed.

**Comment by magicus** [comment]
> I think git does automatically.

Normally perhaps, but now I think you have integrated a PR as a partial commit and then it does not transitively apply contributed-by tags. Or something went awry with the commits since the attribution was missing even there, I could only see it since the commits listed their repo as source.

File: common/src/main/java/com/wynntils/utils/render/RenderUtils.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/utils/render/RenderUtils.java:835]: I am not sure.. most likely a typo. This is why render types are better code wise, porting to them is longer though..

File: gradle.properties (hunk 3 above)
Comment (magicus) [reply gradle.properties]: So are we stuck waiting for Quilt to support 1.20.2?

Comment (kristofbolyai) [reply gradle.properties]: Most likely out today.

**Comment by imIrony** [comment]
Tooltip fitting is kinda wacky

**Comment by kristofbolyai** [comment]
> Tooltip fitting is kinda wacky

Where..? Also, Mc handles it for us now.

**Comment by imIrony** [comment]
> Where..? Also, Mc handles it for us now.

They render half way off the display upwards when they stretch past the screen size

**Comment by kristofbolyai** [comment]
https://github.com/Wynntils/Artemis/pull/2095/commits/c738e290141db9c888f3681b8bdcb4429425f2ae fixes the issues reported by @imIrony.

Could someone review this single commit?

**Review by ShadowCat117 — APPROVED** [review]
Tooltip changes look good!

**Comment by kristofbolyai** [comment]
I am merging this with Quilt disabled. We will bring back quilt support separately.

**Comment by wukibaka** [comment]
> Remember to credit @wukibaka in the final merge.

May I inquire about the reason for crediting me?

**Comment by magicus** [comment]
You submitted PRs to 1.20.1 (which was folded into this PR), I think..?

**Comment by wukibaka** [comment]
> You submitted PRs to 1.20.1 (which was folded into this PR), I think..?

I think I haven't submitted any PRs.

**Comment by magicus** [comment]
I thought github said you helped with the 1.20.1 branch, but maybe I was just confused then. Sorry.

**Pull request merged**